src/osd/osd_types.cc

```diff
@@ -4327,8 +4340,6 @@ void object_info_t::decode(bufferlist::iterator& bl)
     ::decode(omap_digest, bl);
   } else {
     data_digest = omap_digest = -1;
-    clear_flag(FLAG_DATA_DIGEST);
```

Comment:
Isn't clear_flag correct here since we aren't relying on leaving the flag there?